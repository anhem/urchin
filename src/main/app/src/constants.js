export const Actions = {
    Users: {
        GET_USERS: "GET_USERS",
        GET_USERS_SUCCESS: "GET_USERS_SUCCESS"
    },
    User: {
        GET_USER: "GET_USER",
        GET_USER_SUCCESS: "GET_USER_SUCCESS",
        SET_USER: "SET_USER",
        SAVE_USER: "SAVE_USER",
        SAVE_USER_SUCCESS: "SAVE_USER_SUCCESS",
        SAVE_USER_VALIDATION_ERROR: "SAVE_USER_VALIDATION_ERROR",
        DELETE_USER: "DELETE_USER",
        DELETE_USER_SUCCESS: "DELETE_USER_SUCCESS",
        GET_GROUPS_FOR_USER: "GET_GROUPS_FOR_USER",
        GET_GROUPS_FOR_USER_SUCCESS: "GET_GROUPS_FOR_USER_SUCCESS",
        ADD_GROUP: "ADD_GROUP_TO_USER",
        ADD_GROUP_SUCCESS: "ADD_GROUP_TO_USER_SUCCESS",
        REMOVE_GROUP: "REMOVE_GROUP",
        REMOVE_GROUP_SUCCESS: "REMOVE_GROUP_SUCCESS",
    },
    Groups: {
        GET_GROUPS: "GET_GROUPS",
        GET_GROUPS_SUCCESS: "GET_GROUPS_SUCCESS"
    },
    Group: {
        GET_GROUP: "GET_GROUP",
        GET_GROUP_SUCCESS: "GET_GROUP_SUCCESS",
        SET_GROUP: "SET_GROUP",
        SAVE_GROUP: "SAVE_GROUP",
        SAVE_GROUP_SUCCESS: "SAVE_GROUP_SUCCESS",
        SAVE_GROUP_VALIDATION_ERROR: "SAVE_GROUP_VALIDATION_ERROR",
        DELETE_GROUP: "DELETE_GROUP",
        DELETE_GROUP_SUCCESS: "DELETE_GROUP_SUCCESS",
        GET_USERS_FOR_GROUP: "GET_USERS_FOR_GROUP",
        GET_USERS_FOR_GROUP_SUCCESS: "GET_USERS_FOR_GROUP_SUCCESS",
        ADD_USER: "ADD_USER_TO_GROUP",
        ADD_USER_SUCCESS: "ADD_USER_TO_GROUP_SUCCESS",
        REMOVE_USER: "REMOVE_USER",
        REMOVE_USER_SUCCESS: "REMOVE_USER_SUCCESS",
    },
    Folders: {
        GET_FOLDERS: "GET_FOLDERS",
        GET_FOLDERS_SUCCESS: "GET_FOLDERS_SUCCESS"
    },
    Folder: {
        GET_FOLDER: "GET_FOLDER",
        GET_FOLDER_SUCCESS: "GET_FOLDER_SUCCESS",
        SET_FOLDER: "SET_FOLDER",
        SAVE_FOLDER: "SAVE_FOLDER",
        SAVE_FOLDER_SUCCESS: "SAVE_FOLDER_SUCCESS",
        SAVE_FOLDER_VALIDATION_ERROR: "SAVE_FOLDER_VALIDATION_ERROR",
        DELETE_FOLDER: "DELETE_FOLDER",
        DELETE_FOLDER_SUCCESS: "DELETE_FOLDER_SUCCESS",
        SET_CONFIRM_NEW_ENCRYPTED_FOLDER: "SET_CONFIRM_NEW_ENCRYPTED_FOLDER",
        CONFIRM_NEW_ENCRYPTED_FOLDER: "CONFIRM_NEW_ENCRYPTED_FOLDER",
        CONFIRM_NEW_ENCRYPTED_FOLDER_SUCCESS: "CONFIRM_NEW_ENCRYPTED_FOLDER_SUCCESS",
    },
    ACL: {
        GET_ACL: "GET_ACL",
        GET_ACL_SUCCESS: "GET_ACL_SUCCESS",
        UPDATE_GROUP_ACL: "UPDATE_GROUP_ACL",
        UPDATE_GROUP_ACL_SUCCESS: "UPDATE_GROUP_ACL_SUCCESS",
        UPDATE_USER_ACL: "UPDATE_USER_ACL",
        UPDATE_USER_ACL_SUCCESS: "UPDATE_USER_ACL_SUCCESS"
    },
    Site: {
        TOGGLE_NAVIGATION: "TOGGLE_NAVIGATION",
    }
};

export const ErrorCodes = {
    VALIDATION_ERROR: 'VALIDATION_ERROR'
};
