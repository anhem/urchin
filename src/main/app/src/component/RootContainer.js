import React, {Component} from 'react'
import {connect} from "react-redux";
import {isAuthenticated} from "../action/authAction";
import {Actions} from "../constants";
import Root from "./Root";
import {Redirect} from "react-router-dom";

const {Auth} = Actions;

export class RootContainer extends Component {

    componentWillMount() {
        this.props.isAuthenticated();
    }

    render() {
        const authed = this.props.authed;
        let content;

        switch (true) {
            case authed === Auth.LOGIN_SUCCESS:
                content = <Root/>;
                break;
            case authed === Auth.LOGIN_INVALID:
            case authed === Auth.LOGIN_FAILED:
            case authed === Auth.LOGOUT_SUCCESS:
                content = <Redirect to={"/login"}/>;
                break;
            case authed === Auth.PRECONDITION_FAILED:
                content = <Redirect to={"/setup-admin"}/>;
                break;
            default:
                content = <div/>
        }

        return (
            content
        )
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        isAuthenticated: () => {
            dispatch(isAuthenticated())
        }
    }
};

const mapStateToProps = (state) => {
    return {
        authed: state.authData.authed
    }
};

export default connect(mapStateToProps, mapDispatchToProps)(RootContainer)