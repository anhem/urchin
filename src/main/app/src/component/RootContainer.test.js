import React from 'react';
import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json';
import {RootContainer} from "./RootContainer";
import {Actions} from "../constants";

describe('RootContainer', () => {

    it('match snapshot when not authenticated', () => {
        const props = {
            isAuthenticated: jest.fn(),
        };

        expect(toJson(shallow(<RootContainer {...props}/>))).toMatchSnapshot();
    });

    it('match snapshot when logged in', () => {
        const props = {
            isAuthenticated: jest.fn(),
            authed: Actions.Auth.LOGIN_SUCCESS
        };

        expect(toJson(shallow(<RootContainer {...props}/>))).toMatchSnapshot();
    });

    it('match snapshot when login invalid', () => {
        const props = {
            isAuthenticated: jest.fn(),
            authed: Actions.Auth.LOGIN_INVALID
        };

        expect(toJson(shallow(<RootContainer {...props}/>))).toMatchSnapshot();
    });

    it('match snapshot when login failed', () => {
        const props = {
            isAuthenticated: jest.fn(),
            authed: Actions.Auth.LOGIN_FAILED
        };

        expect(toJson(shallow(<RootContainer {...props}/>))).toMatchSnapshot();
    });

    it('match snapshot when precondition failed', () => {
        const props = {
            isAuthenticated: jest.fn(),
            authed: Actions.Auth.PRECONDITION_FAILED
        };

        expect(toJson(shallow(<RootContainer {...props}/>))).toMatchSnapshot();
    });
});
