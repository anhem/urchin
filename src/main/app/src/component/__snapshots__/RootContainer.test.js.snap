// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RootContainer match snapshot when logged in 1`] = `<Root />`;

exports[`RootContainer match snapshot when login failed 1`] = `
<Redirect
  push={false}
  to="/login"
/>
`;

exports[`RootContainer match snapshot when login invalid 1`] = `
<Redirect
  push={false}
  to="/login"
/>
`;

exports[`RootContainer match snapshot when not authenticated 1`] = `<div />`;

exports[`RootContainer match snapshot when precondition failed 1`] = `
<Redirect
  push={false}
  to="/setup-admin"
/>
`;
